datalad/distribution/dataset.py

```diff
@@ -99,6 +102,41 @@ def repo(self):
                 self._repo = AnnexRepo(self._repo.path, create=False)
         return self._repo
 
+    @property
+    def config(self):
+        """Get an instance of the parser for the persistent dataset configuration.
+
+        Returns
+        -------
+        GitConfigParser
+        """
+        if self._cfg is None:
+            cfgpath = opj(self.path, '.datalad', 'config')
+            if not exists(cfgpath):
+                return None
+            self._cfg = git.GitConfigParser(cfgpath, read_only=True)
+        return self._cfg
+
+    @property
+    def config_writer(self):
+        """Get an instance of the parser for modifying the persistent dataset
+        configuration.
+
+        Returns
+        -------
+        GitConfigParser
+        """
+        if self._cfg is None:
+            cfgpath = opj(self.path, '.datalad')
+            if not exists(cfgpath):
+                os.makedirs(cfgpath)
+            self._cfg = git.GitConfigParser(
+                opj(cfgpath, 'config'),
+                read_only=False,
+                merge_includes=False)
+        return self._cfg
```

Comment:
eventually all this custom config handling should be unified I guess so we could harvest configuration settings also from `.git/config` e.g. from sections/options with `datalad-` prefix, similarly to how annex does it.  This way configuration could be persistently (within git) or locally (within .git/config) augmented.  IMHO ;)